exec.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+
+import subprocess
+import time
+import sys
+import os
+import json
+
+
+COMMANDS = ["build", "load-db", "load-db-france", "update-tiles", "clean", "logs", "kartotherian"]
+
+
+def exec_command(command, options):
+    if options.get('debug') is True:
+        print('==> {}'.format(' '.join(command)))
+    p = subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT)
+    while True:
+        output = p.stdout.read1(100)
+        if len(output) == 0:
+            rc = p.poll()
+            if rc is not None:
+                return rc
+        if len(output) > 0:
+            sys.stdout.buffer.write(output)
+            sys.stdout.buffer.flush()
+
+
+def get_submodules():
+    dirs = []
+    try:
+        with open('.gitmodules', 'r') as f:
+            for line in f:
+                line = line.strip()
+                if not line.startswith('path = '):
+                    continue
+                dirs.append(line.split('path = ')[1])
+    except Exception as err:
+        print(f'error happened when trying to get submodules: {err}')
+    return dirs
+
+
+def init_submodule_if_not():
+    submodules = get_submodules()
+    for sub in submodules:
+        if len(os.listdir(sub)) == 0:
+            if exec_command(['git', 'submodule', 'update', '--init', sub]) != 0:
```

Comment:
this command is failing : `exec_command` requires some `options`.  
Why not always calling using `git submodule update --init` to init all submodules ?

Reply:
Because in case you updated the commit, you don't want it to be overwritten if not committed.